domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterSingleton.kt

```diff
@@ -0,0 +1,31 @@
+package org.oppia.android.domain.platformparameter
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+import org.oppia.android.util.platformparameter.PlatformParameter
+
+/** Singleton class which helps in storing and providing Platform Parameters at runtime. */
+@Singleton
+class PlatformParameterSingleton @Inject constructor() {
+  var platformParameterMap: Map<String, ParameterValue> = mapOf()
```

Comment:
It cannot be private, as the inline function (getPlatformParameter) can't acces it. But there can be a workaround like this code snippet - 

      /** Retrieve individual PlatformParameter, if it exists in the platformParameterMap. */
      fun <T : Any> getPlatformParameter(name: String): PlatformParameter<T>? {
        if (platformParameterMap.isEmpty()) return null
        val parameter = platformParameterMap[name] ?: return null
        return object : PlatformParameter<T> {
           override val value: T
            get() = when (getClass<KClass<T>>()) {
              0 -> parameter.string as T
              1 -> parameter.integer as T
              2 -> parameter.boolean as T
              else -> null as T
            }
        }
      }
      
      private inline fun <reified T> getClass(): Int {
         return when (T::class) {
          String::class -> 0
          Int::class -> 1
          Boolean::class -> 2
          else -> 3
       }
      }

Now we can keep the map to be a private var, does this Looks good to you? @jcqli

Reply:
Hm interesting. I've never used inline functions but I would say the original implementation is probably better since the private function isn't adding any code clarity. @vinitamurthi WDYT?

Also, what's the reason for using an inline function?

Reply:
The reason for inline function is to use <reified T>, This is because we want to know which class does T belongs to, which was not possible without using a  <reified T>

Reply:
I did some additional research and I believe I understand now why we have to use the inline function. I'll leave this thread open in case Vinita has additional thoughts.

Reply:
@vinitamurthi PTAL

Reply:
Instead of using an `inline function` and `<reified T>` to get the platform value type, can't we make use of the proto objects here? We already have defined value types through which we can decide which kind of value we'll be returning.

Reply:
We can make use of the Proto classes but doing so will include a bug. For example when we use getPlatformParameter\<String\>("param1") , we expect a parameter with name -> "param1" and a value of String type. But in case we depend on Proto class to set the type of value that is going inside PlatformParameter implementation then this can create a RunTime exception when the Proto has a different data type (than String) stored in the same name of "param1"

Reply:
It looks like we are using the proto objects for type-checking now, is that right?

Reply:
I guess No, we have now made three different functions one each for different data type (String/Int/Boolean). This way we are sure about the type of PlatformParameter to be expected and also in case of any mismatch of the required value type and the cached value type we provide defalt parameter

Reply:
Ah the calls to `parameter.valueTypeCase.equals(ParameterValue.ValueTypeCase.STRING)` are what I mean by using the protos for type-checking, as opposed to using the function generics to get the types. I like this more than the earlier iteration since I thin kit's more aligned with what we want this getter to be, nice job!

Reply:
Thanks.